.github/workflows/build-and-test.yaml

```diff
@@ -0,0 +1,28 @@
+name: build and test
+
+on:
+  push:
+    branches: [production, staging]
+  pull_request:
+    branches: [production, staging]
+
+jobs:
+  build:
+    name: build-test
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18.x'
+          cache: yarn
+      - name: build
+        run: |
+          export CI=false
+          yarn install --check-files
+          yarn build
+      - name: test
+        run: |
+          yarn craco test --watchAll=false --silent
```

Comment:
Don't run tests in watch mode and hide console warnings/errors for ease of reading test results.